### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by mohamed20163858**
in this pull request, I added these functionalities: - 
- the ability to create posts for the currently logged-in user 
-  the ability to create  comments on specific posts for the currently logged-in user 
- the ability to give likes for the currently logged-in user  
and here is a screen shoots for these functionalities: -
image
by clicking on create post button  it will show up on this page and after filling it then click post
image
it will redirect to the posting page 
image
you can write your comments as many as you want it will update automatically after pressing the comment button 
image
also if you click liked button it would disappear after that
image
you could write comments and like other users' posts as well ! 
image
image
image
image

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -5,5 +5,23 @@ def index
...
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author = current_user
+    @post.save
+    redirect_to "/users/#{current_user.id}/posts/#{@post.id}"
+  end
```
Comment (mikethreels) [inline app/controllers/posts_controller.rb]: - [x] in this method you should use the `respond_to` Rails helper method

File: app/controllers/likes_controller.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class LikesController < ApplicationController
+  def create
+    @like = Like.new(like_params)
+    @like.author = current_user
+    @like.save
+    redirect_to "/users/#{flash[:post_author_id]}/posts/#{@like.post_id}"
+  end
```
Comment (mikethreels) [inline app/controllers/likes_controller.rb]: - [x] in this method you should use the `respond_to` Rails helper method

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Mohamed, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (mohamed20163858) [reply app/controllers/posts_controller.rb]: done! thanks alot

File: app/controllers/likes_controller.rb (hunk 2 above)
Comment (mohamed20163858) [reply app/controllers/likes_controller.rb]: done!

**Review by EricMbouwe — APPROVED** [review]
Hi @mohamed20163858, 

Your project is complete! ⚡🔥  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🎉🔥
- Requested changes ✔

## Optional suggestions ☂

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**